src/composables/addressBook.ts

```diff
@@ -0,0 +1,229 @@
+import { computed, ref } from 'vue';
+import { useFileDialog } from '@vueuse/core';
+import { Directory, Filesystem } from '@capacitor/filesystem';
+import { Capacitor } from '@capacitor/core';
+
+import {
+  AccountAddress,
+  IAddressBook,
+  IAddressBookEntry,
+  IAddressBookFilter,
+} from '@/types';
+import { ADDRESS_BOOK_FILTERS, STORAGE_KEYS, MODAL_ADDRESS_BOOK_IMPORT } from '@/constants';
+import { AddressBookEntryExists, AddressBookInvalidAddress, AddressBookRequiredFields } from '@/lib/errors';
+import {
+  generateId,
+  getProtocolByAddress,
+  handleUnknownError,
+  pipe,
+} from '@/utils';
+
+import { createCustomScopedComposable } from './composablesHelpers';
+import { useStorageRef } from './storageRef';
+import { useModals } from './modals';
+
+interface IAddressBookOptions {
+  name: string;
+  address: AccountAddress;
+  isBookmarked?: boolean;
+}
+
+const addressBook = useStorageRef<IAddressBook>({}, STORAGE_KEYS.addressBook);
+
+export const useAddressBook = createCustomScopedComposable(() => {
+  const { openModal, openDefaultModal } = useModals();
+
+  const activeFilter = ref<IAddressBookFilter>(ADDRESS_BOOK_FILTERS.all);
+  const searchQuery = ref<string>('');
+  const showBookmarked = ref(false);
+
+  function filterAddressBookByBookmarked(entries: IAddressBookEntry[]) {
+    return showBookmarked.value ? entries.filter((entry) => entry.isBookmarked) : entries;
+  }
+  function filterAddressBook(entries: IAddressBookEntry[]) {
+    switch (activeFilter.value) {
+      case ADDRESS_BOOK_FILTERS.all:
+        return entries;
+      case ADDRESS_BOOK_FILTERS.bookmarked:
+        return entries.filter((entry) => entry.isBookmarked);
+      default:
+        return activeFilter.value
+          ? entries.filter((entry) => entry.protocol === activeFilter.value)
+          : entries;
+    }
+  }
+  function filterAddressBookBySearchPhrase(entries: IAddressBookEntry[]) {
+    const searchQueryLower = searchQuery.value.toLowerCase();
+    return entries.filter((entry) => {
+      const nameLower = entry.name.toLowerCase();
+      const addressLower = entry.address.toLowerCase();
+      return nameLower.includes(searchQueryLower) || addressLower.includes(searchQueryLower);
+    });
+  }
+  function sortAddressBookByName(entries: IAddressBookEntry[]) {
+    return entries.sort((a, b) => a.name.localeCompare(b.name));
+  }
+
+  const addressBookFiltered = computed(
+    () => pipe([
+      filterAddressBookByBookmarked,
+      filterAddressBook,
+      filterAddressBookBySearchPhrase,
+      sortAddressBookByName,
+    ])(Object.values(addressBook.value)),
+  );
+
+  function setFilter(filter: IAddressBookFilter) {
+    activeFilter.value = filter;
+  }
+  function setShowBookmarked(value: boolean) {
+    showBookmarked.value = value;
+  }
+
+  function addressBookAddressExists(address: AccountAddress) {
```

Comment:
What do you think about renaming this to `isAddressInAddressBook` or `isAddressAddedToAddressBook`?